recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,177 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+required_conan_version = ">=1.31.0"
+
+
+class Mosquitto(ConanFile):
+    name = "mosquitto"
+    license = "EPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mosquitto.org"
+    description = """Eclipse Mosquitto MQTT library, broker and more"""
+    topics = ("MQTT", "IoT", "eclipse")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "ssl": [True, False],
+               "clients": [True, False],
+               "broker": [True, False],
+               "apps": [True, False],
+               "cjson": [True, False],
+               "build_cpp": [True, False],
+               "websockets": [True, False],
+            }
+    default_options = {"shared": False,
+                       "ssl": True,
+                       "clients": False,
+                       "broker": False,
+                       "apps": False,
+                       "cjson": False,
+                       "build_cpp": True,
+                       "websockets": False,
+    }
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.options.clients:
+            del self.options.cjson
+        if not self.options.broker:
+            del self.options.websockets
+        if not self.options.build_cpp:
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.ssl:
+            self.requires("openssl/1.1.1i")
+        if self.options.get_safe("cjson"):
+            self.requires("cjson/1.7.14")
+        if self.options.get_safe("websockets"):
+            self.requires("libwebsockets/4.1.6")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+        self._cmake.definitions["WITH_PIC"] = self.options.get_safe("fPIC", False)
```

Comment:
Best is to set `PIC` to True by default.

On Linux, when using shared libraries, object files **MUST** be compiled with PIC.
Otherwise they can't be used inside a shared library.


```suggestion
        self._cmake.definitions["WITH_PIC"] = self.options.get_safe("fPIC", True)
```

Reply:
can I have a second opinion please, @uilianries , @prince-chrismc  
we do not have an option `fPic` here , did we forget ? 

I am mostly also for always use fPIC, because I recently , but have forgotten where exactly, had some message that I need it.... 

but however, for the libs it seems irrelevant anyway, since I see a lot of things like
```
set_target_properties(libmosquitto PROPERTIES
	POSITION_INDEPENDENT_CODE 1
)
```
in the cmake files of mosquitto, so I guess it's not worth to invalidate the reviews we have ?

Reply:
update: this option, WITH_PIC , is for the static libraries ......

Reply:
True, checking the code again, WITH_PIC is only used for static library. When building shared library, fPIC is activated (hardcoded). Thus, the default value doesn't matter for this recipe.